app/src/main/java/org/stepic/droid/core/presenters/PersistentCourseListPresenter.kt

```diff
@@ -137,7 +137,7 @@ class PersistentCourseListPresenter
                 val allCourses = databaseFacade.getAllCourses(courseType)
 
                 val filteredCourseList: List<Course>
-                if (!applyFilter && !sharedPreferenceHelper.getFilter(courseType).contains(StepikFilter.PERSISTENT)) {
+                if (courseType == Table.enrolled || !applyFilter && !sharedPreferenceHelper.getFilter(courseType).contains(StepikFilter.PERSISTENT)) { // only default filters for enrolled courses
```

Comment:
Also, we have this check here (https://github.com/StepicOrg/stepik-android/pull/149/files#diff-e5bce4da7fc88e784681b99083a1a898R176). I think, that we should extract it.